zulip/integrations/slack/slackdata2zulipdata.py

```diff
@@ -384,8 +386,67 @@ def check_user_mention(text):
     print('ls', os.listdir())
     print('pwd', os.getcwd())
 
-    # TODO
-    # attachments
+    # for attachment.json
+    attachment_json = {}
+    zerver_attachment = []
+
+    # upload folder : zulipdata/uploads/realm_id
+    upload_dir = output_dir + "/uploads/" + str(REALM_ID)
+    # create uploads directory to save all files in that directory
+    os.makedirs(upload_dir)
+
+    # To convert url in form https:\/\/files.slack.com\/files-pri\/..  to https://files.slack.com//files-pri//..
+    def parse_url(url):
+        url = url.replace("\/\/", "//")
+        url = url.replace("\/", "//")
+        return url
+
+    # Download from the slack url and save it to the folder zulipdata/uploads/realm_id/id/filename
+    def save_attachment(url, id, name):
+        url = parse_url(url)
+        response = requests.get(url, stream=True)
```

Comment:
This would work only if the Slack URL is world-readable though. Which I don't think is the case.

Reply:
Maybe there should be a `try` and print a warning message if the files can't be downloaded.

Reply:
The download works in Travis though. So I assume for public Slack instances, these are world-readable.

Reply:
So I'll the ` try` here?

Reply:
Yes. If an exception happens due to permission denied, then just stop constructing the attachment, and let it be in TOODOO for PM stuff.